components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayJMSMessageListener.java

```diff
@@ -111,24 +114,39 @@ private void handleNotificationMessage(String eventType, long timestamp, String
         if ((APIConstants.EventType.DEPLOY_API_IN_GATEWAY.name().equals(eventType)
                 || APIConstants.EventType.REMOVE_API_FROM_GATEWAY.name().equals(eventType))
                 && gatewayArtifactSynchronizerProperties.isRetrieveFromStorageEnabled()) {
-            DeployAPIInGatewayEvent gatewayEvent = new Gson().fromJson(new String(eventDecoded),
-                    DeployAPIInGatewayEvent.class);
+            DeployAPIInGatewayEvent gatewayEvent = new Gson().fromJson(new String(eventDecoded), DeployAPIInGatewayEvent.class);
             gatewayEvent.getGatewayLabels().retainAll(gatewayArtifactSynchronizerProperties.getGatewayLabels());
             if (!gatewayEvent.getGatewayLabels().isEmpty()) {
                 String gatewayLabel = gatewayEvent.getGatewayLabels().iterator().next();
-                try {
-                    if (APIConstants.EventType.DEPLOY_API_IN_GATEWAY.name().equals(eventType)) {
-                        inMemoryApiDeployer.deployAPI(gatewayEvent.getApiId(), gatewayLabel);
-                        if (debugEnabled) {
-                            log.debug(gatewayEvent.getEventId() + " processed and deployed in gateway");
+                Runnable task = null;
+                if (APIConstants.EventType.DEPLOY_API_IN_GATEWAY.name().equals(eventType)) {
+                    task = new Runnable() {
+
+                        @Override public void run() {
+                            try {
+                                inMemoryApiDeployer.deployAPI(gatewayEvent.getApiId(), gatewayLabel);
+                            } catch (ArtifactSynchronizerException e) {
+                                log.error("Error in deploying artifacts");
```

Comment:
Don't we have to include context information in the error log? ie: API id

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/9542/commits/63c98e787dbe2323e2fc0775fc72e5c5d9c7f030